python_modules/libraries/dagster-dg/dagster_dg/cli/env.py

```diff
@@ -42,3 +46,70 @@ def list_env_command(**global_options: object) -> None:
         table.add_row(key, value)
     console = Console()
     console.print(table)
+
+
+# ########################
+# ##### PULL
+# ########################
+
+
+def _get_local_secrets_for_locations(
+    client: DagsterCloudGraphQLClient, location_names: set[str]
+) -> Mapping[str, Mapping[str, str]]:
+    secrets_by_location = {location_name: {} for location_name in location_names}
+
+    result = client.execute(LOCAL_SECRETS_FILE_QUERY)
+    for secret in result["secretsOrError"]["secrets"]:
+        if not secret["localDeploymentScope"]:
+            continue
```

Comment:
it seems bad to fetch a bunch of non locally scoped secrets over the network and then filter them out in memory afterwards, can we add a serverside filter for this?

Reply:
https://github.com/dagster-io/internal/pull/14646